include/sparrow/mp_utils.hpp

```diff
@@ -258,5 +260,23 @@ namespace sparrow::mpl
         return find_if(list, predicate::same_as<TypeToFind>{});
     }
 
+    template <template <typename...> class, typename>
+    inline constexpr bool is_instantiation_of_v = false;
+    template <template <typename...> class C, typename... T>
+    inline constexpr bool is_instantiation_of_v<C, C<T...>> = true;
+    /// @returns `true` if `T` is an instantiation of `C`.
+    template <template <typename...> class C, typename... T>
+    concept InstantiationOf = is_instantiation_of_v<C, T...>;
```

Comment:
Name convention is snake_case for concepts here.

Note also that there is already `is_type_instance_of/_v` in the same file. It's a type trait and a value but concepts are often better defined through type traits first.

Reply:
replaced by `is_type_instance_of/v` 👍